src/cryptonote_core/CMakeLists.txt

```diff
@@ -59,6 +59,7 @@ target_link_libraries(cryptonote_core
     ringct
     device
     hardforks
+    rpc_pub
```

Comment:
Don't like this. IMO we should avoid introducing this dependency.

Reply:
Agreed. If it's a pain to avoid I could live with it though.

Reply:
Using `boost::function` (not `std::function`, see comments). The penalty is an indirect function call instead of a direct function call when `zmq_pub` is enabled. Its also less obvious now  that the call is doing a `shared_ptr` lock, but the code for blockchain notification has the same issue.